src/main/kotlin/com/terraformation/backend/nursery/api/WithdrawalsController.kt

```diff
@@ -0,0 +1,114 @@
+package com.terraformation.backend.nursery.api
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonSetter
+import com.fasterxml.jackson.annotation.Nulls
+import com.terraformation.backend.api.NurseryEndpoint
+import com.terraformation.backend.api.SuccessResponsePayload
+import com.terraformation.backend.db.default_schema.FacilityId
+import com.terraformation.backend.db.nursery.BatchId
+import com.terraformation.backend.db.nursery.WithdrawalId
+import com.terraformation.backend.db.nursery.WithdrawalPurpose
+import com.terraformation.backend.nursery.db.BatchStore
+import com.terraformation.backend.nursery.model.BatchWithdrawalModel
+import com.terraformation.backend.nursery.model.ExistingWithdrawalModel
+import com.terraformation.backend.nursery.model.NewWithdrawalModel
+import io.swagger.v3.oas.annotations.media.ArraySchema
+import java.time.LocalDate
+import javax.validation.constraints.Min
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@NurseryEndpoint
+@RequestMapping("/api/v1/nursery/withdrawals")
+@RestController
+class WithdrawalsController(
+    val batchStore: BatchStore,
+) {
+  @PostMapping
+  fun createBatchWithdrawal(
+      @RequestBody payload: CreateNurseryWithdrawalRequestPayload
+  ): CreateNurseryWithdrawalResponsePayload {
+    val model = batchStore.withdraw(payload.toModel())
+    val batchModels = model.batchWithdrawals.map { batchStore.fetchOneById(it.batchId) }
+
+    return CreateNurseryWithdrawalResponsePayload(
+        batchModels.map { BatchPayload(it) },
+        NurseryWithdrawalPayload(model),
+    )
+  }
+}
+
+data class BatchWithdrawalPayload(
+    val batchId: BatchId,
+    @JsonSetter(nulls = Nulls.FAIL) @Min(0) val germinatingQuantityWithdrawn: Int,
+    @JsonSetter(nulls = Nulls.FAIL) @Min(0) val notReadyQuantityWithdrawn: Int,
+    @JsonSetter(nulls = Nulls.FAIL) @Min(0) val readyQuantityWithdrawn: Int,
```

Comment:
I added some comments in here but my review draft doesn't show it anymore. If these comments are redundant apologies..
A few questions.
Based on designs, none of the withdrawal scenarios equire specifying the germinating quantity - is this mostly looking ahead and providing full coverage?
Outplant withdrawal seems to ask for a 'withdrawn quantity' - should that be ready quantity (client usage)?

Reply:
> Based on designs, none of the withdrawal scenarios equire specifying the germinating quantity - is this mostly looking ahead and providing full coverage?

Yeah, pretty much. It seems wrong to me that there's no way to, e.g., tell the system you've thrown away dead seeds that failed to germinate. But you raise a good point and I think the right fix is to make this default to 0 rather than being required, so the client can leave it out of the payloads to start with.

> Outplant withdrawal seems to ask for a 'withdrawn quantity' - should that be ready quantity (client usage)?

PRD says withdrawals always ask for not-ready and ready quantities, so I'm not 100% sure of the intent, but your thought seems like it's probably correct.

Reply:
> > Based on designs, none of the withdrawal scenarios equire specifying the germinating quantity - is this mostly looking ahead and providing full coverage?
> 
> Yeah, pretty much. It seems wrong to me that there's no way to, e.g., tell the system you've thrown away dead seeds that failed to germinate. But you raise a good point and I think the right fix is to make this default to 0 rather than being required, so the client can leave it out of the payloads to start with.
> 
> > Outplant withdrawal seems to ask for a 'withdrawn quantity' - should that be ready quantity (client usage)?
> 
> PRD says withdrawals always ask for not-ready and ready quantities, so I'm not 100% sure of the intent, but your thought seems like it's probably correct.

Got it, thanks :) i should cross check with the PRD instead of designs when reviewing here. Re: design/PRD differences  - I will bring them up with design outside the scope of this review.